app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt

```diff
@@ -0,0 +1,155 @@
+package com.android.doctorapp.ui.doctor
+
+import android.util.Log
+import android.util.Patterns
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.android.doctorapp.R
+import com.android.doctorapp.di.ResourceProvider
+import com.android.doctorapp.di.base.BaseViewModel
+import com.android.doctorapp.repository.AddDoctorRepository
+import com.android.doctorapp.repository.models.ApiErrorResponse
+import com.android.doctorapp.repository.models.ApiNoNetworkResponse
+import com.android.doctorapp.repository.models.ApiSuccessResponse
+import com.android.doctorapp.repository.models.UserDataRequestModel
+import com.android.doctorapp.util.SingleLiveEvent
+import com.android.doctorapp.util.extension.asLiveData
+import com.android.doctorapp.util.extension.isEmailAddressValid
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class AddDoctorViewModel @Inject constructor(
+    private val resourceProvider: ResourceProvider,
+    private val addRepository: AddDoctorRepository
+) : BaseViewModel() {
+
+    val doctorName: MutableLiveData<String?> = MutableLiveData()
+    val doctorNameError: MutableLiveData<String?> = MutableLiveData()
+
+    val doctorEmail: MutableLiveData<String> = MutableLiveData()
+    val doctorEmailError: MutableLiveData<String?> = MutableLiveData()
+
+    val doctorContactNumber: MutableLiveData<String> = MutableLiveData()
+    val doctorContactNumberError: MutableLiveData<String?> = MutableLiveData()
+
+    val toggleLiveData: MutableLiveData<Boolean> = MutableLiveData(false)
+
+    private var auth: FirebaseAuth? = null
+
+    val isDataValid: MutableLiveData<Boolean> = MutableLiveData(false)
+
+    private val _navigationListener = SingleLiveEvent<Int>()
+    val navigationListener = _navigationListener.asLiveData()
+
+    private val _addDoctorResponse = SingleLiveEvent<String>()
+    val addDoctorResponse = _addDoctorResponse.asLiveData()
+
+    private lateinit var fireStore: FirebaseFirestore
+    private lateinit var firebaseUser: FirebaseUser
+
+    init {
```

Comment:
Please define all primary type object define  in baseViewModel. So any View Model can access those objects.

Reply:
Done